### PR TITLE
Use implicit-hie when no explicit hie.yaml

### PR DESCRIPTION
**Pull request by jneira**
* The test suite hangs in my windows 10 😟 
* I've tried several projects without hie.yaml and seems to work fine

//cc @Avi-D-coder

### CODE REVIEW

**Comment by pepeiborra** [comment]
See my comment in https://github.com/haskell/haskell-language-server/issues/386#issuecomment-690620411
What does `loadImplicitHieCradle` do for projects that have no Cabal/Stack descriptors?

File: ghcide.cabal [hunk 1]
```diff
@@ -201,10 +202,6 @@ library
         Development.IDE.GHC.HieAst
         Development.IDE.GHC.HieBin
     ghc-options: -Wall -Wno-name-shadowing -Wincomplete-uni-patterns
-    -- This is needed to prevent a GHC crash when building
-    -- Development.IDE.Session with stack on 8.10.1 on Windows
-    if (impl(ghc > 8.9) && os(windows))
-      ghc-options: -fexternal-interpreter
```
Comment (fendor) [inline ghcide.cabal:207]: Why has this been removed?

**Review by fendor — APPROVED** [review]
LGTM! Thank you!

EDIT: I have just read the comment, and I think this is correct, gen-hie works for cabal and stack projects, only

**Comment by fendor** [comment]
We maybe need more code-paths to support stand-alone files.

File: ghcide.cabal (hunk 1 above)
Comment (jneira) [reply ghcide.cabal:207]: Sorry for nor mentioning it, my local windows build was failing due to a segfault in the external interpreter (the thing that line try to fix in ci!) 
The ci job has failied for other reason, let's see if it continue failing due to segfault without `-fexternal-interpreter` after fixin the actual error.

**Comment by jneira** [comment]
> 
> 
> We maybe need more code-paths to support stand-alone files.

jumm i assumed incorrectly gen-hie would keep the fallback to the direct cradle with no build system present. I'll take a look.

**Comment by jneira** [comment]
My original plan was use `loadImplicitHieCradle` and fallback to actual hie-bios `loadImplicitCradle` in case the first one fails with an error.
But i think we decided in other pr to keep only one path, maybe it is time to reconsider it.

**Comment by fendor** [comment]
Can we turn `loadImplicitHieCradle` into `loadImplicitHieCradle :: FilePath -> IO (Maybe (Cradle a))`? E.g. only return a cradle if it is stack or cabal? If it is Nothing, we can default to hie-bios implicit cradle.

**Comment by jneira** [comment]
Well, i am not sure if `hie-implicit-cradle` does not handle standlone files (although `gen-hie` does not for sure):
```
D:\ws\haskell\standalone-hs>ghcide
ghcide version: 0.3.0 (GHC: 8.8) (PATH: D:\bin\ghcide.exe)
Ghcide setup tester in D:\ws\haskell\standalone-hs.
Report bugs at https://github.com/digital-asset/ghcide/issues

Step 1/4: Finding files to test in D:\ws\haskell\standalone-hs
Found 1 files

Step 2/4: Looking for hie.yaml files that control setup
Found 1 cradle
  ()

Step 3/4: Initializing the IDE

Step 4/4: Type checking the files
[INFO] Consulting the cradle for "D:\\ws\\haskell\\standalone-hs\\Main.hs"
Output from setting up the cradle Cradle {cradleRootDir = "D:\\ws\\haskell\\standalone-hs", cradleOptsProg = CradleAction: Default}
[INFO] Using interface files cache dir: C:\Users\user\AppData\Local\ghcide\main-da39a3ee5e6b4b0d3255bfef95601890afd80709
[INFO] Making new HscEnv[main]

Completed (1 file worked, 0 files failed)

D:\ws\haskell\standalone-hs>ghcide-impl-hie
ghcide version: 0.3.0 (GHC: 8.8) (PATH: D:\bin\ghcide-impl-hie.exe)
Ghcide setup tester in D:\ws\haskell\standalone-hs.
Report bugs at https://github.com/digital-asset/ghcide/issues

Step 1/4: Finding files to test in D:\ws\haskell\standalone-hs
Found 1 files

Step 2/4: Looking for hie.yaml files that control setup
Found 1 cradle
  ()

Step 3/4: Initializing the IDE

Step 4/4: Type checking the files
[INFO] Consulting the cradle for "D:\\ws\\haskell\\standalone-hs\\Main.hs"
Output from setting up the cradle Cradle {cradleRootDir = "D:\\ws\\haskell\\standalone-hs", cradleOptsProg = CradleAction: Default}
[INFO] Using interface files cache dir: C:\Users\user\AppData\Local\ghcide\main-da39a3ee5e6b4b0d3255bfef95601890afd80709
[INFO] Making new HscEnv[main]

Completed (1 file worked, 0 files failed)
```
`ghcide` is HEAD and `ghcide-impl-hie` is this pr
I will test it with vscode too but maybe @Avi-D-coder could confirm it

**Comment by pepeiborra** [comment]
I think this is great and I'm happy to merge it, but I have a couple of requests that can either be done in this PR of in follow-ups:

1. Inform the user that there is no cradle #788 
2. Write the implicit cradle to disk 
3. Give the user a choice if there are both stack and cabal project descriptors
4. Delete all the cradle files from ghcide and hls - no longer needed

**Review by pepeiborra — APPROVED** [review]

**Review by fendor — APPROVED** [review]
LGTM, the rest can be adressed in follow-up PRs

**Comment by jneira** [comment]
> 
> 
> I think this is great and I'm happy to merge it, but I have a couple of requests that can either be done in this PR of in follow-ups:
> 
>     1. Inform the user that there is no cradle #788
> 
>     2. Write the implicit cradle to disk
> 
>     3. Give the user a choice if there are both stack and cabal project descriptors
> 
>     4. Delete all the cradle files from ghcide and hls - no longer needed

Agree with all of them, i would add a fifth one: state in the (at least) the hls readme the cases when the implicit cradle will fail: common stanzas and conditionals that introduce new modules+path definitions. I think (hope?) they are not so frequent but still.

**Pull request merged**